test/powershell/engine/ETS/Adapter.Tests.ps1

```diff
@@ -1,143 +1,228 @@
+                    return $self.CallCounter
+                }
 
-        class TestCodeMethodInvokationWithVoidReturn {
-            [int] $CallCounter
+                static [void] VoidMethodCM([PSObject] $self) {
+                    $self.CallCounter++
+                }
 
-            static [int] IntMethodCM([PSObject] $self) {
-                return $self.CallCounter
+                static [Reflection.MethodInfo] GetMethodInfo([string] $name) {
+                    return [TestCodeMethodInvokationWithVoidReturn].GetMethod($name)
+                }
             }
 
-            static [void] VoidMethodCM([PSObject] $self) {
-                $self.CallCounter++
-            }
+            Update-TypeData -Force -TypeName TestCodeMethodInvokationWithVoidReturn -MemberType CodeMethod -MemberName IntMethod -Value ([TestCodeMethodInvokationWithVoidReturn]::GetMethodInfo('IntMethodCM'))
+            Update-TypeData -Force -TypeName TestCodeMethodInvokationWithVoidReturn -MemberType CodeMethod -MemberName VoidMethod -Value ([TestCodeMethodInvokationWithVoidReturn]::GetMethodInfo('VoidMethodCM'))
+            try {
+                $o = [TestCodeMethodInvokationWithVoidReturn]::new()
+                $o.CallCounter | Should Be 0
+                $o.VoidMethod()
+                $o.CallCounter | Should be 1
 
-            static [Reflection.MethodInfo] GetMethodInfo([string] $name) {
-                return [TestCodeMethodInvokationWithVoidReturn].GetMethod($name)
+                $o.IntMethod() | Should be 1
+            }
+            finally {
+                Remove-TypeData TestCodeMethodInvokationWithVoidReturn
             }
         }
 
-        Update-TypeData -Force -TypeName TestCodeMethodInvokationWithVoidReturn -MemberType CodeMethod -MemberName IntMethod -Value ([TestCodeMethodInvokationWithVoidReturn]::GetMethodInfo('IntMethodCM'))
-        Update-TypeData -Force -TypeName TestCodeMethodInvokationWithVoidReturn -MemberType CodeMethod -MemberName VoidMethod -Value ([TestCodeMethodInvokationWithVoidReturn]::GetMethodInfo('VoidMethodCM'))
-        try {
-            $o = [TestCodeMethodInvokationWithVoidReturn]::new()
-            $o.CallCounter | Should Be 0
-            $o.VoidMethod()
-            $o.CallCounter | Should be 1
+        It "Count and length property works for singletons" {
+            # Return magic Count and Length property if it absent.
+            $x = 5
+            $x.Count | Should Be 1
+            $x.Length | Should Be 1
 
-            $o.IntMethod() | Should be 1
-        }
-        finally {
-            Remove-TypeData TestCodeMethodInvokationWithVoidReturn
+            $null.Count | Should Be 0
+            $null.Length | Should Be 0
+
+            (10).Count | Should Be 1
+            (10).Length | Should Be 1
+
+            ("a").Count | Should Be 1
+            # The Length property exists for strings, so we skip the test in the context.
+            # ("a").Length | Should Be 1
```

Comment:
Maybe better to have the test but change the comment that `Length` should use the member for `[string]` type.  In which case, it would be better to test with "aa".

Reply:
Added test and changed the comment.